ChangeLog

```diff
@@ -6,10 +6,41 @@ Pydot versions since 1.4.2 adhere to [PEP 440-style semantic versioning]
 (https://www.python.org/dev/peps/pep-0440/#semantic-versioning).
 
 
-3.0.3 (unreleased)
+4.0.0 (unreleased)
 ------------------
 
-- Nothing changed yet.
+Added:
+- Added initial type annotations. (#430)
+  There are still a lot of `Any`s, but Pydot now officially supports type annotations.
+
+Changed:
+- Simplify `Common.get_top_graph_type` with a default. (#437)
+- Define parser elements in a separate `GraphParser` class. (#464)
+  It's a user-transparent change, but now it's easier to debug, test, and modify the parser.
+  See the related Pull Request for more information.
+
+Fixed:
+- Avoid subprocess window creation within `call_graphviz` on Windows. (#431)
+- Removed a blank line inserted after subgraphs in the `to_string()` output. (#434)
+- Fixed "all numeric" quoting behavior. (#439)
+- Capitalized keywords in labels are now quoted. (#449)
+- Fixed the issue where the parser breaks off lines at hash symbol following a HTML tag (#451)
+- Fixed complex edge parsing. (#467)
```

Comment:
I'd even say this could be an "added", maybe? Our parser has support for subgraph edge endpoints now. (It gave the _appearance_ of supporting them before, but the support was less "broken" than "completely absent".)